sdk/metric/meter.go

```diff
@@ -201,28 +208,115 @@ func (m *meter) Float64ObservableGauge(name string, options ...instrument.Float6
 // RegisterCallback registers the function f to be called when any of the
 // insts Collect method is called.
 func (m *meter) RegisterCallback(f metric.Callback, insts ...instrument.Asynchronous) (metric.Registration, error) {
+	if len(insts) == 0 {
+		// Don't allocate a observer if not needed.
+		return noopRegister{}, nil
+	}
+
+	reg := newObserver()
+	var errs multierror
 	for _, inst := range insts {
-		// Only register if at least one instrument has a non-drop aggregation.
-		// Otherwise, calling f during collection will be wasted computation.
-		switch t := inst.(type) {
-		case *instrumentImpl[int64]:
-			if len(t.aggregators) > 0 {
-				return m.registerMultiCallback(f)
+		switch o := inst.(type) {
```

Comment:
This switch case won't allow a delegated instrument.  If you create an instrument prior to `global.SetMeterProvider()`, that instrument can't be used in a `mtr.RegisterCallback()` after the set.

eg: https://gist.github.com/MadVikingGod/79945fa262d3e13c2908aab3d5ad59e3

Reply:
Gotcha :+1:. Updating.